src/backend/dx12/src/device.rs

```diff
@@ -572,34 +693,76 @@ impl Device {
-            image::Kind::Cube(..) |
-            image::Kind::CubeArray(..) => d3d12::D3D12_UAV_DIMENSION_TEXTURE3D,
-        };
+        #![allow(non_snake_case)]
+        assert_eq!(range.levels.start + 1, range.levels.end);
 
         let mut desc = d3d12::D3D12_UNORDERED_ACCESS_VIEW_DESC {
             Format: format,
-            ViewDimension: dimension,
+            ViewDimension: 0,
             u: unsafe { mem::zeroed() },
         };
 
+        let MipSlice = range.levels.start as _;
+        let FirstArraySlice = range.layers.start as _;
+        let ArraySize = (range.layers.end - range.layers.start) as _;
+
         match kind {
-            image::Kind::D2(_, _, _) => {
+            image::Kind::D1(_) => {
+                assert_eq!(range.layers, 0..1);
+                desc.ViewDimension = d3d12::D3D12_UAV_DIMENSION_TEXTURE1D;
+                *unsafe{ desc.u.Texture1D_mut() } = d3d12::D3D12_TEX1D_UAV {
+                    MipSlice,
+                }
+            }
+            image::Kind::D1Array(_, num_layers) => {
+                assert!(range.layers.end <= num_layers);
+                desc.ViewDimension = d3d12::D3D12_UAV_DIMENSION_TEXTURE1DARRAY;
+                *unsafe{ desc.u.Texture1DArray_mut() } = d3d12::D3D12_TEX1D_ARRAY_UAV {
+                    MipSlice,
+                    FirstArraySlice,
+                    ArraySize,
+                }
+            }
+            image::Kind::D2(_, _, image::AaMode::Single) => {
+                assert_eq!(range.layers, 0..1);
+                desc.ViewDimension = d3d12::D3D12_UAV_DIMENSION_TEXTURE2D;
                 *unsafe{ desc.u.Texture2D_mut() } = d3d12::D3D12_TEX2D_UAV {
-                    MipSlice: range.levels.start as _,
-                    PlaneSlice: 0,
+                    MipSlice,
+                    PlaneSlice: 0, //TODO
                 }
             }
-            _ => unimplemented!()
+            image::Kind::D2Array(_, _, num_layers, image::AaMode::Single) => {
+                assert!(range.layers.end <= num_layers);
+                desc.ViewDimension = d3d12::D3D12_UAV_DIMENSION_TEXTURE2DARRAY;
+                *unsafe{ desc.u.Texture2DArray_mut() } = d3d12::D3D12_TEX2D_ARRAY_UAV {
+                    MipSlice,
+                    FirstArraySlice,
+                    ArraySize,
+                    PlaneSlice: 0, //TODO
+                }
+            }
+            image::Kind::D2(..) |
+            image::Kind::D2Array(..) => {
+                error!("MSAA images can't be viewed as UAV");
+                return Err(image::ViewError::Unsupported);
+            }
+            image::Kind::D3(_, _, _) => {
+                assert_eq!(range.layers, 0..1);
+                desc.ViewDimension = d3d12::D3D12_UAV_DIMENSION_TEXTURE3D;
+                *unsafe{ desc.u.Texture3D_mut() } = d3d12::D3D12_TEX3D_UAV {
+                    MipSlice,
+                    FirstWSlice: FirstArraySlice, //TODO: double-check
```

Comment:
You already have set the assert correct :D 
FirstWSlice should be 0 and WSize can be -1

Reply:
I see, will fix